soc/arm/silabs_exx32/common/soc_gpio.h

```diff
@@ -25,12 +25,6 @@ struct soc_gpio_pin {
 	unsigned int out;       /** out register value */
 };
 
-/**
- * @brief Configure GPIO pin
- * @param[in] pin configuration data
- */
-void soc_gpio_configure(const struct soc_gpio_pin *pin);
```

Comment:
where is `soc_gpio_pin` struct used now? I guess it can be deleted, and drivers include `<em_gpio.h>`

Reply:
This struct is used in drivers that are compatible with other SiLabs boards that use non-pinctrl mode for GPIO configuration, see for example [UART driver](https://github.com/antmicro/zephyr/blob/efr32bg22-initial/drivers/serial/uart_gecko.c#L326-L329) or [SPI driver](https://github.com/antmicro/zephyr/blob/efr32bg22-initial/drivers/spi/spi_gecko.c#L191-L196) (other drivers are: `i2c` and `eth`). I've seen similar solutions for [atmel_sam](https://github.com/antmicro/zephyr/blob/efr32bg22-initial/soc/arm/atmel_sam/common/soc_gpio.h#L85-L94) and [cypress](https://github.com/antmicro/zephyr/blob/efr32bg22-initial/soc/arm/cypress/common/soc_gpio.h#L67-L71) SoCs. I think it is a good idea to leave this structure because it allows us to unclutter driver configuration structures from 4 different values that are specific to GPIO configuration (GPIO port, pin, mode, out register value) per each GPIO pin used in the driver (see [UART config struct](https://github.com/antmicro/zephyr/blob/efr32bg22-initial/drivers/serial/uart_gecko.c#L89-L90)).